pvlib/iotools/panond.py

```diff
@@ -0,0 +1,197 @@
+    key1 = value1
+      level2 = second level
+      key2 = value2'
+
+    output:
+
+    {
+    level1: first level
+    key1: value1,
+    level2:{
+        level2: second level,
+        key2: value2
+        }
+    }
+
+    The parser takes an additional step to infer the datatype present in
+    each value. The .pan/.ond files appear to have intentially left datatype
+    indicators (e.g. floats have '.' decimals). However, there is still the
+    possibility that the datatype applied from this parser is incorrect. In
+    that event the user would need to convert to the desired datatype.
+
+    See Also
+    --------
+
+    References
+    ----------
+    """
+    comp = {}  # Component
+    dict_levels = [comp]
+
+    fbuf.seek(0)
+    lines = fbuf.getvalue().splitlines()
+
+    for i in range(0, len(lines) - 1):
+        if lines[i] == '':  # Skipping blank lines
+            continue
+        # Reading blank lines. Stopping one short to avoid index error.
+        # Last line never contains important data.
+        # Creating variables to assist new level in dictionary creation logic
+        indent_lvl_1 = (len(lines[i]) - len(lines[i].lstrip(' '))) // 2
+        indent_lvl_2 = (len(lines[i + 1]) - len(lines[i + 1].lstrip(' '))) // 2
+        # Split the line into key/value pair
+        line_data = lines[i].split('=')
+        key = line_data[0].strip()
+        # Logical to make sure there is a value to extract
+        if len(line_data) > 1:
+            value = element_type(line_data[1].strip())
+        else:
+            value = None
+        # add a level to the dict. If a key/value pair triggers the new level,
+        # the key/value will be repeated in the new dict level.
+        # Not vital to file function.
+        if indent_lvl_2 > indent_lvl_1:
+            current_level = dict_levels[indent_lvl_1]
+            new_level = {}
+            current_level[key] = new_level
+            dict_levels = dict_levels[: indent_lvl_1 + 1] + [new_level]
+            current_level = dict_levels[indent_lvl_1 + 1]
+            current_level[key] = value
+
+        elif indent_lvl_2 <= indent_lvl_1:  # add key/value to dict
+            current_level = dict_levels[indent_lvl_1]
+            current_level[key] = value
+
+    return comp
+
+
+def read_panond(file):
+    """
+    Retrieve Module or Inverter data from a .pan or .ond text file,
+    respectively.
+
+    Parameters
+    ----------
+    file : string or path object
+        Name or path of a .pan/.ond file
+
+    Returns
+    -------
+    content : Nested Dictionary
+        Contents of the .pan or .ond file following the indentation of the
+        file. The value of datatypes are assumed during reading. The value
+        units are the default used by PVsyst.
+
+    Raises
+    ------
+
+    Notes
+    -----
+    The read function simply converts a file path to a file-like object and
+    passes it to the parser. At time of creation, tested .pan/.ond files used
```

Comment:
```suggestion
    The read_panond function converts a file path to a file-like object,
    passes it to parse_panond, and returns the file content. At time of creation,
    tested .pan/.ond files used
```

Reply:
@cwhanse I made the proposed grammar/spelling changes. I am content moving forward with the changes as they are now.

Reply:
Would you like some help with the tests? I would push to your fork, but the PR is made from your main branch so I can't do that.